cleanlab_pinecone_rag_workflow/Cleanlab_Pinecone_RAG.ipynb

```diff
@@ -0,0 +1,2222 @@
+      "Metadata: {'filename': 'documents/Authorization_for_Direct_Deposit_IFSP.pdf', 'quality': 'good', 'topic': 'finance'}\n",
+      "---\n",
+      "Document: Blackstone Third Quarter 2023 Investor Call October 19, 2023 at 9:00am ETModerator: Good day and wel...\n",
+      "Metadata: {'filename': 'documents/Blackstone-Third-Quarter-2023-Investor-Call.pdf', 'quality': 'good', 'topic': 'finance'}\n",
+      "---\n"
+     ]
+    }
+   ],
+   "source": [
+    "results = rag_pipeline.search(\n",
+    "    query=\"Your search query\",\n",
+    "    top_k=2,\n",
+    "    filter_query={\n",
+    "        \"topic\": {\"$eq\": \"finance\"},\n",
+    "        \"quality\": {\"$eq\": \"good\"}\n",
+    "    }\n",
+    ")\n",
+    "\n",
+    "for doc_text, metadata in results:\n",
+    "    print(f\"Document: {doc_text[:100]}...\")\n",
+    "    print(f\"Metadata: {metadata}\")\n",
+    "    print(\"---\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c50fa034-b6de-4e14-b7c0-f446888874a8",
+   "metadata": {},
+   "outputs": [],
+   "source": []
+  },
+  {
+   "cell_type": "markdown",
+   "id": "941afdc4-f0d3-4bd4-b2c2-cab8ac545a50",
+   "metadata": {},
+   "source": [
+    "### View Retreived Document"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 50,
+   "id": "379aa3bf-c488-46ff-8c4a-6a52e6e49c89",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Top doc for this query: \n",
+      "\n",
+      "('Welcome to the topic on the sales process.In this topic, we will examine the basic steps in the sales process in SAP Business One. Then we will perform a streamlined sales process.2Imagine that your company has set customer satisfaction as its top priority. You review the basic steps in the sales process and decide to use a streamlined process for rush orders to ensure that customer demands can be met as quickly as possible.3(cid:132) The sales-A/R process moves from selling the goods (and services) to delivering those goods to invoicing the customer for their purchases, to the last step in accounting of receiving the incoming payment.(cid:132) Each step in the sales process involves a document, such as a sales order, delivery, or A/R invoice. SAP Business One efficiently moves all relevant information from one document to the next in the document flow. According to the business needs, you can choose which documents to include as your process steps. The only mandatory document in sales', {'filename': 'documents/sap-sales-overview.pdf', 'quality': 'unknown', 'topic': 'sales'})\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Simple query regarding our documents\n",
+    "question = \"Tell me about the sales principles at SAP Business One\"\n",
+    "\n",
+    "top_doc = rag_pipeline.search(question, top_k=1, filter_query={\"topic\": {\"$eq\": \"sales\"}})\n",
```

Comment:
for this part, you should write some text before the code cell to justify where the metadata filter is coming from. Eg:

Suppose our application asks this user which topic their question is about from a pre-defined list. Alternatively, we could train a classifier to predict the topic from the question (automatically using Cleanlab Studio AutoML).